libraries/chain/include/eosio/chain/backing_store/db_combined.hpp

```diff
@@ -209,76 +194,128 @@ class rocksdb_whole_db_table_collector {
 
    template<typename IndexType>
    void extract_secondary_index(b1::chain_kv::bytes::const_iterator remaining,
                                 b1::chain_kv::bytes::const_iterator key_end, const char *value,
                                 std::size_t value_size) {
-      IndexType secondary_key;
-      uint64_t primary_key;
-      EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, secondary_key), bad_composite_key_exception,
-                 "DB intrinsic key-value store composite key is malformed");
-      EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, primary_key), bad_composite_key_exception,
-                 "DB intrinsic key-value store composite key is malformed");
-      backing_store::payer_payload pp(value, value_size);
-      receiver_.add_row(secondary_index_view<IndexType>{primary_key, pp.payer, secondary_key});
+      if (!is_table_only()) {
+         IndexType secondary_key;
+         uint64_t primary_key;
+         EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, secondary_key), bad_composite_key_exception,
+                    "DB intrinsic key-value store composite key is malformed");
+         EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, primary_key), bad_composite_key_exception,
+                    "DB intrinsic key-value store composite key is malformed");
+         backing_store::payer_payload pp(value, value_size);
+         receiver_.add_row(secondary_index_view<IndexType>{primary_key, pp.payer, secondary_key});
+      }
       ++secondary_count<IndexType>();
    }
 
    bool operator()(uint64_t contract, const char *key, std::size_t key_size,
                    const char *value, std::size_t value_size) {
-      if (!keep_processing_()) {
-         return false;
-      }
       b1::chain_kv::bytes composite_key(key, key + key_size);
       auto[scope, table, remaining, type] = backing_store::db_key_value_format::get_prefix_thru_key_type(
             composite_key);
 
+      const name code = name{contract};
+      if (table_context_ && is_reversed() && !is_same_table(code, scope, table)) {
+         table_context_->count = total_count();
+         receiver_.add_row(*table_context_);
+      }
+
+      if (!keep_processing_()) {
+         // performing the check after retrieving the prefix to provide similar results for RPC calls
+         // (indicating next scope)
+         stopped_processing_ = true;
+         table_context_ = table_id_object_view{code, scope, table, name{}, 0};
+         return false;
+      }
+
       if (type == key_type::table) {
          backing_store::payer_payload pp(value, value_size);
 
-         auto add_row = [this](const auto &row) { receiver_.add_row(row); };
 
-         uint32_t total_count = primary_count_;
-         for (auto& count : all_secondary_indices_count_) {
-            total_count += count;
-            count = 0;
+         if (is_reversed()) {
+            table_context_ = table_id_object_view{code, scope, table, pp.payer, 0};
```

Comment:
Add the following comment:
// since table is reversed, we see the table key first, so we have the whole table context, except for the count of keys contained in the table, that will need to be added later once the next table is identified (so that we know that this table is complete)

Reply:
Added.